std/uni.d

```diff
@@ -2835,7 +2837,7 @@ private:
             //  [---+++----++++----++++++]
             //  [                         a  b]
             data.append(a, b);
-            return data.length-1;
+            return data.length > 0 ? data.length - 1 : 0;
```

Comment:
Does this fix a bug? If so, can we get a unittest for it? If not, would `assert(data.length > 0);` be enough?